### PR TITLE
Expand system_request.cc ut coverage

### PR DESCRIPTION
**Pull request by BSolonenko**
This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
This PR provides full unit test coverage for `system_request.cc` unit

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by BSolonenko** [comment]
@AKalinich-Luxoft @AByzhynar @LuxoftAKutsan @aderiabin @Ypostolov @ValeriiMalkov Please review.

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 1]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
 
   ExpectManageMobileCommandWithResultCode(mobile_apis::Result::REJECTED);
 
-  SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(msg));
-  command->Run();
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest,
+       Run_HttpRequestTypeAndExistsFileName_SuccessResponce) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko propose to rename to `Run_HttpRequestTypeAndFileExists_SuccessResponse`

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 2]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+  file.is_download_complete = true;
+  EXPECT_CALL(*mock_app_, GetFile(app_full_file_path)).WillOnce(Return(&file));
+
+  EXPECT_CALL(mock_policy_handler_, ReceiveMessageFromSDK(kFileName, _));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::SUCCESS);
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest, Run_RequestType_SuccessRequest) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko rename to `Run_ClimateRequestType_SuccessResponse`

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 3]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+  EXPECT_CALL(*mock_app_, GetFile(app_full_file_path)).WillOnce(Return(&file));
+
+  EXPECT_CALL(app_mngr_,
+              ManageHMICommand(HMIResultCodeIs(
+                  hmi_apis::FunctionID::BasicCommunication_SystemRequest)));
+
+  Run(CreateIVSUMessage(kRequestClimate));
+}
+
+TEST_F(SystemRequestTest, Run_NonExistentApp_ApplicationNotRegisteredResponse) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko rename to `Run_AppDoesntExist_ApplicationNotRegisteredResponse)`

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 4]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+      .WillOnce(Return(mock_app_));
+
+  ExpectManageMobileCommandWithResultCode(
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest,
+       Run_EmptyParams_DisallowedRequestType_DisallowedResponse) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko rename to `Run_DisallowedRequestType_DisallowedResponse`

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 5]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(false));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::DISALLOWED);
+
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest, Run_InvalidFileNameSyntax_InvalidData) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko `Run_InvalidFileName_InvalidDataResponse`

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 6]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::INVALID_DATA);
+
+  MessageSharedPtr message = CreateIVSUMessage();
+  (*message)[am::strings::msg_params][am::strings::file_name] =
+      kFileName + "\n";
+
+  Run(message);
+}
+
+TEST_F(SystemRequestTest, Run_InvalidFileName_InvalidData) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko I believe this unit test covers the same code as `Run_InvalidFileNameSyntax_InvalidData`. Isn't it?

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 7]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko is it possible to move this expectation to `PreConditions()`? As I could see, these two calls are always together.

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 8]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+      .WillOnce(Return(true));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::INVALID_DATA);
+
+  MessageSharedPtr message = CreateIVSUMessage();
+  (*message)[am::strings::msg_params][am::strings::file_name] = kFileName + "/";
+
+  Run(message);
+}
+TEST_F(SystemRequestTest, Run_InvalidSaveBinary_GenericError) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko `Run_InvalidBinaryData_GenericErrorResponse`

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 9]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::INVALID_DATA);
+
+  MessageSharedPtr message = CreateIVSUMessage();
+  (*message)[am::strings::msg_params][am::strings::file_name] = kFileName + "/";
+
+  Run(message);
+}
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko please split all test cases with a blank line

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 10]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestQueryApps))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(app_mngr_, SaveBinary(_, kSystemFilesPath, kFileName, 0))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(app_mngr_, application_by_policy_id(_))
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko is this param unknown? Maybe `kAppPolicyId`?

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 11]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::SUCCESS);
+
+  MessageSharedPtr message = CreateIVSUMessage();
+  Event event(hmi_apis::FunctionID::BasicCommunication_SystemRequest);
+  event.set_smart_object(*message);
+  SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(message));
+  command->on_event(event);
+}
+
+TEST_F(SystemRequestTest, OnEvent_NonExistentApp_NullPointer) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko `OnEvent_AppDoesNotExist_NullPointerError`

**Comment by AByzhynar** [comment]
@BSolonenko What is `system_reques` ??? In PR name.

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 12]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
   }
 
+  void Run(MessageSharedPtr message) {
+    SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(message));
+    ASSERT_TRUE(command->Init());
+    command->Run();
+  }
+
+  void SetUp() OVERRIDE {
+    file_system::CreateDirectoryRecursively(binary_data_folder);
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko If directories creation is critical - add checks for it

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 13]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+    file_system::CreateDirectoryRecursively(kSystemFilesPath + "/");
+    file_system::CreateFile(app_full_file_path);
+    file_system::CreateFile(kSystemFilesPath + "/" + kFileName);
+  }
+
+  void TearDown() OVERRIDE {
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko Check sections

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 14]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+TEST_F(SystemRequestTest, OnEvent_BasicCommunicationSystemRequest_Success) {
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::SUCCESS);
+
+  MessageSharedPtr message = CreateIVSUMessage();
+  Event event(hmi_apis::FunctionID::BasicCommunication_SystemRequest);
+  event.set_smart_object(*message);
+  SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(message));
+  command->on_event(event);
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko Init & Run must be invoked before on_event

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 15]
```diff
@@ -68,23 +71,35 @@ const std::string kAppPolicyId = "fake-app-id";
 const uint32_t kDefaultTimeout = 1000u;
 const std::string kAppFolderName = "fake-app-name";
 const std::string kAppStorageFolder = "fake-storage";
-const std::string kSystemFilesPath = "/fake/system/files";
+const std::string kSystemFilesPath = "fake/system/files";
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko why do we need so deep folder structure? Please rename to "fake-system-files"

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 16]
```diff
@@ -68,23 +71,35 @@ const std::string kAppPolicyId = "fake-app-id";
...
 
 class SystemRequestTest
     : public CommandRequestTest<CommandsTestMocks::kIsNice> {
  public:
-  SystemRequestTest() : mock_app_(CreateMockApp()) {}
+  SystemRequestTest()
+      : mock_app_(CreateMockApp())
+      , binary_data_folder(kAppStorageFolder + "/" + kAppFolderName + "/")
+      , app_full_file_path(binary_data_folder + kFileName)
+      , binary_data_(10, 10) {}
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko is is very not obvious that  binary_data_ will be filled with 10 values of 10 .
Please use fill value something like 0xFF.

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 17]
```diff
@@ -68,23 +71,35 @@ const std::string kAppPolicyId = "fake-app-id";
...
  protected:
-  MessageSharedPtr CreateIVSUMessage() {
-    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
-    (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
-    (*msg)[am::strings::msg_params][am::strings::request_type] =
-        mobile_apis::RequestType::HTTP;
-    (*msg)[am::strings::msg_params][am::strings::file_name] = kFileName;
-    return msg;
+  MessageSharedPtr CreateIVSUMessage(
+      const mobile_apis::RequestType::eType request_type = kRequestHttpType) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko please remove default argument. It is rather confusing

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 18]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
                             am::commands::Command::CommandOrigin::ORIGIN_SDL));
   }
 
+  void Run(MessageSharedPtr message) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko such code is widely used in multiple command tests. Please move it to upper level of abstraction.

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 19]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+
+  void TearDown() OVERRIDE {
+    file_system::RemoveDirectory(kAppStorageFolder);
+    file_system::RemoveDirectory(kSystemFilesPath);
+  }
+
   MockAppPtr mock_app_;
   MockPolicyHandlerInterface mock_policy_handler_;
+  const std::string binary_data_folder;
+  const std::string app_full_file_path;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko please handle class variable naming style

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 20]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+  std::vector<uint8_t> binary_data_;
 };
 
 TEST_F(SystemRequestTest, Run_HTTP_FileName_no_binary_data_REJECTED) {
   PreConditions();
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko can you move preconditions to Setup() ?

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 21]
```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
...
+}
+TEST_F(SystemRequestTest, Run_InvalidSaveBinary_GenericError) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(app_mngr_,
+              SaveBinary(binary_data_, kSystemFilesPath, kFileName, 0))
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @BSolonenko I don't see any significatn use of binary data. Can we remove it from class members and keep on stack of test cases?

**Pull request closed without merging**